### DISCUSSION TITLE
i18n issue,

### DISCUSSION THREAD

**Discussion by futurediffusion** [lobehub/sd-webui-lobe-theme#270] [General]
Hello friends, I made a translation in Spanish, and I put it in the lang folder, I also modified the config and index files of the i18n folder. but the extension does not recognize it in the section to choose languages. I am doing something wrong? Or do I need to modify something else? I would also like to be able to share the es_ES file with the creator so that he can add it by default.

**Comment by canisminor1990** [comment]
just refactor it and  move the locales to https://github.com/canisminor1990/sd-webui-lobe-theme/tree/main/locales

you can add {lng}.json inside it then add lng options to [options.json](https://github.com/canisminor1990/sd-webui-lobe-theme/blob/main/locales/options.json)

don't need to rebuild js client, the locales files will server by python url api

**Comment by canisminor1990** [comment]
welcome pr 😚

**Reply by futurediffusion** [reply]
const translation = {
  community: 'Comunidad',
  custom: 'Personalizado',
  extraNetwork: 'Red Extra',
  feedback: 'Comentarios',
  fixed: 'Fijo',
  float: 'Flotante',
  help: 'Ayuda',
  kitchen: 'Cocina',
  loadPrompt: 'Cargar Instrucción',
  lobe: 'Lóbulo',
  moreProducts: 'Más Productos',
  negative: 'Negativo',
  positive: 'Positivo',
  quickSetting: 'Configuración Rápida',
  resizable: 'Redimensionable',
  resources: 'Recursos',
  scroll: 'Desplazar',
  setPrompt: 'Establecer Instrucción',
  setting: 'Configuración',
  settingButtomReset: 'Restablecer',
  settingButtomSubmit: 'Aplicar y Reiniciar Interfaz',
  settingCustomFont: 'Cargar Fuente Personalizada',
  settingCustomFontDesc:
    'Cuando está habilitado, cargará automáticamente una webfont para mejorar la visualización de texto en chino, inglés y código',
  settingCustomLogo: 'Logotipo Personalizado',
  settingCustomLogoDesc: 'Soporta URL / Base64 / Símbolos Emoji',
  settingCustomTitle: 'Título Personalizado',
  settingCustomTitleDesc: 'Título del Logotipo Personalizado',
  settingExtraNetworkSidebarDefaultCardSize: 'Tamaño de Portada de Modelo',
  settingExtraNetworkSidebarDefaultCardSizeDesc: 'Valor predeterminado del tamaño de portada de modelo al iniciar',
  settingExtraNetworkSidebarDefaultExpand: 'Expansión Predeterminada',
  settingExtraNetworkSidebarDefaultExpandDesc:
    'Si se debe expandir la barra lateral de forma predeterminada al iniciar',
  settingExtraNetworkSidebarDefaultWidth: 'Ancho Predeterminado',
  settingExtraNetworkSidebarDefaultWidthDesc: 'Ancho predeterminado de la barra lateral al iniciar',
  settingExtraNetworkSidebarDisplayMode: 'Modo de Visualización',
  settingExtraNetworkSidebarDisplayModeDesc:
    'Modo de cuadrícula fijo para visualización constante, se expande automáticamente cuando el mouse se mueve hacia el lateral en modo flotante',
  settingExtraNetworkSidebarEnable: 'Habilitar',
  settingExtraNetworkSidebarEnableDesc: 'Habilitar la barra lateral de red extra en el lado derecho',
  settingGroupExtraNetworkSidebar: 'Barra Lateral de Red Extra',
  settingGroupLayout: 'Configuración de Diseño',
  settingGroupPromotTextarea: 'Área de Texto de Instrucción',
  settingGroupQuickSettingSidebar: 'Barra Lateral de Configuración Rápida',
  settingGroupTheme: 'Configuración de Tema',
  settingHideFooter: 'Ocultar Pie de Página',
  settingHideFooterDesc:
    'Ocultar el pie de página del tema y solo mostrar el pie de página predeterminado de Stable Diffusion WebUI',
  settingLanguage: 'Idioma',
  settingLanguageDesc: 'Idioma del tema Lóbulo',
  settingLogoPreview: 'Vista Previa',
  settingLogoType: 'Tipo de Logotipo',
  settingLogoTypeDesc: 'Tipo de Logotipo',
  settingNeutralColor: 'Color Neutro',
  settingNeutralColorDesc:
    'Personaliza diferentes tonos de gris con diferentes tendencias de color, el segundo es el color neutro original de Kitchen',
  settingPrimaryColor: 'Color Primario',
  settingPrimaryColorDesc:
    'Color primario personalizado, el segundo es el color original del tema Kitchen',
  settingPromptDisplayMode: 'Modo de Visualización',
  settingPromptDisplayModeDesc: 'Altura fija o altura automática con soporte de redimensionamiento arrastrable',
  settingPromptEditor: 'Editor de Instrucción',
  settingPromptEditorDesc: 'Proporciona un editor de instrucciones sencillo en la parte superior de la barra lateral de configuración rápida',
  settingPromptHighlight: 'Resaltado de Sintaxis de Instrucción',
  settingPromptHighlightDesc:
    'Colorea automáticamente la visualización de la instrucción de acuerdo con las reglas de sintaxis de Stable Diffusion',
  settingQuickSettingSidebarDefaultExpand: 'Expansión Predeterminada',
  settingQuickSettingSidebarDefaultExpandDesc:
    'Si se debe expandir la barra lateral de forma predeterminada al iniciar',
  settingQuickSettingSidebarDefaultWidth: 'Ancho Predeterminado',
  settingQuickSettingSidebarDefaultWidthDesc: 'Ancho predeterminado de la barra lateral al iniciar',
  settingQuickSettingSidebarDisplayMode: 'Modo de Visualización',
  settingQuickSettingSidebarDisplayModeDesc:
    'Modo de cuadrícula fijo para visualización constante, se expande automáticamente cuando el mouse se mueve hacia el lateral en modo flotante',
  settingQuickSettingSidebarEnable: 'Habilitar',
  settingQuickSettingSidebarEnableDesc: 'Habilitar la barra lateral de configuración rápida en el lado izquierdo',
  settingReduceAnimation: 'Reducir Animación',
  settingReduceAnimationDesc:
    'Reduce el efecto de desenfoque y el color de flujo de fondo, lo que puede mejorar la suavidad y ahorrar uso de CPU',
  settingSplitPreviewer: 'Dividir Visor',
  settingSplitPreviewerDesc:
    'Coloca el cuadro de entrada de instrucciones a la izquierda y el botón de generación a la derecha, asegurando que la imagen generada siempre se muestre en la parte superior al desplazarse (experimental)',
  settingSvgIcons: 'Usar Iconos SVG',
  settingSvgIconsDesc: 'Reemplazar todos los iconos Emoji en Stable Diffusion WebUI con iconos SVG globalmente',
  switchTheme: 'Cambiar Tema Claro/Oscuro',
  sync: 'Sincronizar con la configuración de webui',
  themeFeedback: 'Comentarios de Tema',
  themeSetting: 'Configuración de Tema',
};

export default translation;

**Reply by futurediffusion** [reply]
here its the translate , if you want to add it, by default.

**Comment by dhwz** [comment]
@canisminor1990 
the last commit is broken
```
*** Error loading script: settings.py
    Traceback (most recent call last):
      File "/home/dragon/stable-diffusion-webui/modules/scripts.py", line 295, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "/home/dragon/stable-diffusion-webui/modules/script_loading.py", line 10, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "/home/dragon/stable-diffusion-webui/extensions/sd-webui-lobe-theme/scripts/settings.py", line 11, in <module>
        from scripts.lib.api import LobeApi
      File "/home/dragon/stable-diffusion-webui/extensions/sd-webui-lobe-theme/scripts/lib/api.py", line 8, in <module>
        from scripts.lib.locale import LobeLocale
    ModuleNotFoundError: No module named 'scripts.lib.locale'

---
```

**Comment by canisminor1990** [comment]
fixed

**Comment by dhwz** [comment]
@canisminor1990 
now is this broken

```
*** Error loading script: settings.py
    Traceback (most recent call last):
      File "/home/dragon/stable-diffusion-webui/modules/scripts.py", line 295, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "/home/dragon/stable-diffusion-webui/modules/script_loading.py", line 10, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "/home/dragon/stable-diffusion-webui/extensions/sd-webui-lobe-theme/scripts/settings.py", line 10, in <module>
        from scripts.lib.lobe_log import LobeLog
      File "/home/dragon/stable-diffusion-webui/extensions/sd-webui-lobe-theme/scripts/lib/lobe_log.py", line 19, in <module>
        LobeLog = LobeLogClass()
      File "/home/dragon/stable-diffusion-webui/extensions/sd-webui-lobe-theme/scripts/lib/lobe_log.py", line 9, in __init__
        self.logging_enabled = shared.cmd_opts.lobe_debug
    AttributeError: 'Namespace' object has no attribute 'lobe_debug'

---
```
